builtin/providers/azurerm/resource_arm_resource_group.go

```diff
@@ -0,0 +1,162 @@
+package azurerm
+
+import (
+	"fmt"
+	"log"
+	"regexp"
+	"strings"
+	"time"
+
+	"github.com/hashicorp/azure-sdk-for-go/arm/resources"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+// resourceArmResourceGroup returns the *schema.Resource
+// associated to resource group resources on ARM.
+func resourceArmResourceGroup() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmResourceGroupCreate,
+		Read:   resourceArmResourceGroupRead,
+		Exists: resourceArmResourceGroupExists,
+		Delete: resourceArmResourceGroupDelete,
+
+		Schema: map[string]*schema.Schema{
+			"name": &schema.Schema{
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validateArmResourceGroupName,
+			},
+			"location": &schema.Schema{
+				Type:      schema.TypeString,
+				Required:  true,
+				ForceNew:  true,
+				StateFunc: azureRMNormalizeLocation,
+			},
+		},
+	}
+}
+
+// validateArmResourceGroupName validates inputs to the name argument against the requirements
+// documented in the ARM REST API guide: http://bit.ly/1NEXclG
+func validateArmResourceGroupName(v interface{}, k string) (ws []string, es []error) {
+	value := v.(string)
+
+	if len(value) > 80 {
+		es = append(es, fmt.Errorf("%q may not exceed 80 characters in length", k))
+	}
+
+	if strings.HasSuffix(value, ".") {
+		es = append(es, fmt.Errorf("%q may not end with a period", k))
+	}
+
+	if matched := regexp.MustCompile(`[\(\)\.a-zA-Z0-9_-]`).Match([]byte(value)); !matched {
+		es = append(es, fmt.Errorf("%q may only contain alphanumeric characters, dash, underscores, parentheses and periods", k))
+	}
+
+	return
+}
+
+// resourceArmResourceGroupCreate goes ahead and creates the specified ARM resource group.
+func resourceArmResourceGroupCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient)
+	resGroupClient := client.resourceGroupClient
+
+	name := d.Get("name").(string)
+	location := d.Get("location").(string)
+
+	log.Printf("[INFO] Issuing Azure ARM creation request for resource group '%s'.", name)
+
+	rg := resources.ResourceGroup{
+		Name:     &name,
+		Location: &location,
+	}
+
+	_, err := resGroupClient.CreateOrUpdate(name, rg)
+	if err != nil {
+		return fmt.Errorf("Error issuing Azure ARM create request for resource group '%s': %s", name, err)
+	}
+
+	d.SetId(*rg.Name)
```

Comment:
The response resources.ResourceGroup's name should be the same as the `name` provided from the ResourceData here...
If you look closely, you'll notice that all the `struct`s in the SDK also have a string `ID` field, which [if I remember correctly,] are given back by ARM in the response.
Now, this is not to say that it is wrong to internally track resources after their schema `name`, however, the `ID`s of said resources will be required when they are referenced in other resources (see the SDK's SubResource `struct`ures...).

I'd thus suggest that we either go ahead and take note of the `ID` in a `Computed: true,` field so that users may reference the resource in their code, or use said ID as the schema ID of the resource...